elastic/integrations
# apm: add transaction.profiler_stack_trace_ids field to traces datastream mapping

## Proposed commit message



Add `transaction.profiler_stack_trace_ids field` to traces datastream mapping

_It requires the **package-spec v3.1.0** which should be shipped with the **next release of elastic-package**(current release is v0.96.1)_

## Checklist


## Author's Checklist


- [x] Wait for the new release of `elastic-package` which includes package version 3.1.0

## How to test this PR locally


clone the [elastic-package](https://github.com/elastic/elastic-package), then run:
```sh
❯ make clean
...
❯ make build
...
❯ make install
```
This will move the binary built from the latest commit to `$GOBIN`.

Then go to the apm package directory and checkout to this branch, then run:
```sh
❯ elastic-package check
...
❯ elastic-package lint
``` 
It shouldn't print any error logs, and look something like this:
<img width="951" alt="image" src="https://github.com/elastic/integrations/assets/16660866/5a2c2f06-d665-4898-ae67-06e63834e9c7">

### Test with Registry
```
❯ elastic-package stack up -v -d --version 8.13.0-SNAPSHOT
# once the stack is up(registry must be running), install the package
# integrations/packages/apm
❯ elastic-package build
...
❯ elastic-package install
```

When installed, Go to Kibana, and navigate to `Stack Management > Index Management > Component` and search for `traces`. the component should show the added field with the counted_keyword type.

<img width="1512" alt="image" src="https://github.com/elastic/integrations/assets/16660866/bbd7a286-64b8-4bf3-98c4-49575697fbd3">

## Related issues


- Relates https://github.com/elastic/apm-data/pull/202 
- Relates https://github.com/elastic/apm-server/issues/12054
- Relates https://github.com/elastic/elasticsearch/pull/105223
